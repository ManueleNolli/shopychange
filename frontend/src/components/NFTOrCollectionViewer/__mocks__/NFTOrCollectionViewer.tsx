import { divWithoutChildrenMock } from '../../../../mocksUtils'

const NFTOrCollectionViewerMock = () =>
  divWithoutChildrenMock('NFTOrCollectionViewer')
export default NFTOrCollectionViewerMock
