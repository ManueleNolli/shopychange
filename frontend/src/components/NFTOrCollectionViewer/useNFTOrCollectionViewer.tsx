import { useState } from 'react'

export default function useNFTOrCollectionViewer() {
  const [nftsView, setNftsView] = useState(true) // true = NFTs, false = Collections

  return {
    nftsView,
    setNftsView,
  }
}
