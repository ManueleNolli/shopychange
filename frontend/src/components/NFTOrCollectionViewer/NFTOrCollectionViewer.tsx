import React from 'react'
import { Box, Button, Center, Text } from '@chakra-ui/react'
import { NFTOrCollectionViewerProps } from '../../types/props/NFTOrCollectionViewerProps'
import NFTCardGrid from '../NFTCardGrid/NFTCardGrid'
import CollectionGrid from '../CollectionGrid/CollectionGrid'

import { MdViewModule } from 'react-icons/md'
import { MdViewList } from 'react-icons/md'
import useNFTOrCollectionViewer from './useNFTOrCollectionViewer'

export default function NFTOrCollectionViewer(
  props: NFTOrCollectionViewerProps
) {
  const { nftsView, setNftsView } = useNFTOrCollectionViewer() // true = NFTs, false = Collections

  const renderNFTs = () => {
    if (props.nfts.length === 0) return
    if (nftsView) {
      return <NFTCardGrid {...props} />
    } else {
      return <CollectionGrid {...props} />
    }
  }

  return (
    <Box mt="2vh" w="100%">
      {props.nfts.length === 0 ? (
        <Box w="full">
          <Center>
            <Text py="10">No NFTs found</Text>
          </Center>
        </Box>
      ) : (
        <Box
          h="5vh"
          w="100%"
          px={5}
          justifyContent={'flex-end'}
          alignItems={'center'}
          display={'flex'}
        >
          <Button isActive={nftsView} onClick={() => setNftsView(true)} mr={2}>
            <MdViewModule
              size={'70%'}
              style={{ marginRight: '2px' }}
              data-testid="button-nfts"
            />
            NFTs
          </Button>
          <Button
            isActive={!nftsView}
            onClick={() => setNftsView(false)}
            ml={2}
          >
            <MdViewList
              size={'60%'}
              style={{ marginRight: '2px' }}
              data-testid="button-collections"
            />
            Collections
          </Button>
        </Box>
      )}

      {renderNFTs()}
    </Box>
  )
}
