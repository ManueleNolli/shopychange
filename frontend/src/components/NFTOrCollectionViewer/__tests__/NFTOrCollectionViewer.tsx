import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import NFTOrCollectionViewer from '../NFTOrCollectionViewer'

jest.mock('../../NFTCardGrid/NFTCardGrid')

jest.mock('../../CollectionGrid/CollectionGrid')

const mockData = [
  {
    contractAddress: '0x0000000000',
    tokenId: 0,
    image:
      'https://ipfs.io/ipfs/bafybeietntip4sq3xmuwiz3s43yb3klrpa3okgtg5deysmbd5hbizhrb2q/0.png',
    name: 'Test NFT',
  },
  {
    contractAddress: '0x0000000000',
    tokenId: 1,
    image:
      'https://ipfs.io/ipfs/bafybeietntip4sq3xmuwiz3s43yb3klrpa3okgtg5deysmbd5hbizhrb2q/1.png',
    name: 'Test NFT 1',
  },
]
describe('NFTOrCollectionViewer', () => {
  test('renders NFTOrCollectionViewer with NFT option correctly', () => {
    const tree = render(<NFTOrCollectionViewer nfts={mockData} />)

    expect(tree.getByTestId('NFTCardGrid')).toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })

  test('renders NFTOrCollectionViewer with Collection option correctly', () => {
    const tree = render(<NFTOrCollectionViewer nfts={mockData} />)

    const button = tree.getByTestId('button-collections')
    fireEvent.click(button)

    expect(tree.getByTestId('CollectionGrid')).toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })

  test('return to NFT view', () => {
    const tree = render(<NFTOrCollectionViewer nfts={mockData} />)

    const buttonCollections = tree.getByTestId('button-collections')
    fireEvent.click(buttonCollections)

    const buttonNFTs = tree.getByTestId('button-nfts')
    fireEvent.click(buttonNFTs)

    expect(tree.getByTestId('NFTCardGrid')).toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })

  test('renders successfully with empty data', () => {
    const tree = render(<NFTOrCollectionViewer nfts={[]} />)

    expect(tree.getByText('No NFTs found')).toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })
})
