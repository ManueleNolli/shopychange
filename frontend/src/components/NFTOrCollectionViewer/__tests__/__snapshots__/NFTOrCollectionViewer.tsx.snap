// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFTOrCollectionViewer renders NFTOrCollectionViewer with Collection option correctly 1`] = `
<div>
  <div
    class="css-deelig"
  >
    <div
      class="css-40oaw"
    >
      <button
        class="chakra-button css-1emije3"
        type="button"
      >
        <svg
          data-testid="button-nfts"
          fill="currentColor"
          height="70%"
          stroke="currentColor"
          stroke-width="0"
          style="margin-right: 2px;"
          viewBox="0 0 24 24"
          width="70%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M14.67 5v6.5H9.33V5h5.34zm1 6.5H21V5h-5.33v6.5zm-1 7.5v-6.5H9.33V19h5.34zm1-6.5V19H21v-6.5h-5.33zm-7.34 0H3V19h5.33v-6.5zm0-1V5H3v6.5h5.33z"
          />
        </svg>
        NFTs
      </button>
      <button
        class="chakra-button css-18zcl9e"
        data-active=""
        type="button"
      >
        <svg
          data-testid="button-collections"
          fill="currentColor"
          height="60%"
          stroke="currentColor"
          stroke-width="0"
          style="margin-right: 2px;"
          viewBox="0 0 24 24"
          width="60%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 14h4v-4H3v4zm0 5h4v-4H3v4zM3 9h4V5H3v4zm5 5h13v-4H8v4zm0 5h13v-4H8v4zM8 5v4h13V5H8z"
          />
        </svg>
        Collections
      </button>
    </div>
    <div
      data-testid="CollectionGrid"
    />
  </div>
</div>
`;

exports[`NFTOrCollectionViewer renders NFTOrCollectionViewer with NFT option correctly 1`] = `
<div>
  <div
    class="css-deelig"
  >
    <div
      class="css-40oaw"
    >
      <button
        class="chakra-button css-1emije3"
        data-active=""
        type="button"
      >
        <svg
          data-testid="button-nfts"
          fill="currentColor"
          height="70%"
          stroke="currentColor"
          stroke-width="0"
          style="margin-right: 2px;"
          viewBox="0 0 24 24"
          width="70%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M14.67 5v6.5H9.33V5h5.34zm1 6.5H21V5h-5.33v6.5zm-1 7.5v-6.5H9.33V19h5.34zm1-6.5V19H21v-6.5h-5.33zm-7.34 0H3V19h5.33v-6.5zm0-1V5H3v6.5h5.33z"
          />
        </svg>
        NFTs
      </button>
      <button
        class="chakra-button css-18zcl9e"
        type="button"
      >
        <svg
          data-testid="button-collections"
          fill="currentColor"
          height="60%"
          stroke="currentColor"
          stroke-width="0"
          style="margin-right: 2px;"
          viewBox="0 0 24 24"
          width="60%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 14h4v-4H3v4zm0 5h4v-4H3v4zM3 9h4V5H3v4zm5 5h13v-4H8v4zm0 5h13v-4H8v4zM8 5v4h13V5H8z"
          />
        </svg>
        Collections
      </button>
    </div>
    <div
      data-testid="NFTCardGrid"
    />
  </div>
</div>
`;

exports[`NFTOrCollectionViewer renders successfully with empty data 1`] = `
<div>
  <div
    class="css-deelig"
  >
    <div
      class="css-13rbvqk"
    >
      <div
        class="css-gmuwbf"
      >
        <p
          class="chakra-text css-9452eu"
        >
          No NFTs found
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`NFTOrCollectionViewer return to NFT view 1`] = `
<div>
  <div
    class="css-deelig"
  >
    <div
      class="css-40oaw"
    >
      <button
        class="chakra-button css-1emije3"
        data-active=""
        type="button"
      >
        <svg
          data-testid="button-nfts"
          fill="currentColor"
          height="70%"
          stroke="currentColor"
          stroke-width="0"
          style="margin-right: 2px;"
          viewBox="0 0 24 24"
          width="70%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M14.67 5v6.5H9.33V5h5.34zm1 6.5H21V5h-5.33v6.5zm-1 7.5v-6.5H9.33V19h5.34zm1-6.5V19H21v-6.5h-5.33zm-7.34 0H3V19h5.33v-6.5zm0-1V5H3v6.5h5.33z"
          />
        </svg>
        NFTs
      </button>
      <button
        class="chakra-button css-18zcl9e"
        type="button"
      >
        <svg
          data-testid="button-collections"
          fill="currentColor"
          height="60%"
          stroke="currentColor"
          stroke-width="0"
          style="margin-right: 2px;"
          viewBox="0 0 24 24"
          width="60%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 14h4v-4H3v4zm0 5h4v-4H3v4zM3 9h4V5H3v4zm5 5h13v-4H8v4zm0 5h13v-4H8v4zM8 5v4h13V5H8z"
          />
        </svg>
        Collections
      </button>
    </div>
    <div
      data-testid="NFTCardGrid"
    />
  </div>
</div>
`;
